prov/efa/src/rdm/efa_rdm_ope.c

```diff
@@ -1513,15 +1512,7 @@ int efa_rdm_ope_post_remote_write(struct efa_rdm_ope *ope)
 		if (ep->efa_outstanding_tx_ops == ep->efa_max_outstanding_tx_ops)
 			return -FI_EAGAIN;
 
-		if (!ope->desc[iov_idx]) {
-			/* efa_rdm_ope_try_fill_desc() did not fill the desc,
-			 * which means memory registration failed.
-			 * return -FI_EAGAIN here will cause user to run progress
-			 * engine, which will cause some memory registration
-			 * in MR cache to be released.
-			 */
-			return -FI_EAGAIN;
-		}
+		assert(ope->desc[iov_idx]);
```

Comment:
Hm, I think if application calls write without a desc that's illegal, but I'm not sure it's an assert, should probably be a FI_EINVAL?

Reply:
Another concern is for inject_write, which doesn't have a desc passed by application. We currently use emulated write for that but I am concerned that if we in the future choose to use rdma write for inject we will miss this pitfall.

Reply:
+1 f or Luke, here we should return FI_EINVAL as it's not a valid user case. assert should be used to validate something we have guaranteed in our own code.